elastic/kibana
# [Response Ops] Conditionally run background tasks based on `node.roles` configuration

Resolves https://github.com/elastic/kibana/issues/133551

## Summary

Uses the new `node.roles` configuration in `kibana.yml` to determine whether this instance of Kibana task manager should poll for tasks. If task polling is disabled, this Kibana will reject ephemeral tasks and the Task Manager health API will not return the `runtime` response field.

## To Verify

1. Run Kibana as is and create a rule that runs frequently. Verify that the rule runs as expected. Access the Task Manager health API at `/api/task_manager/_health` and verify you can see the runtime response field at `stats.runtime`
2. Update `kibana.yml` to disable task polling using the config: `node.roles: ['ui']`
3. Notice that your rule stops running and doesn't get picked up even tho the `runAt` field has expired. Access the Task Manager health API at `/api/task_manager/_health` and verify you can no longer see the runtime response field at `stats.runtime`
4. Try out different `node.roles` configs to see that the results are as expected (other options are `node.roles: ['*']` and `node.roles: ['background_tasks']` but neither of these should disable task polling.


### Checklist
